File: client/src/app/modules/quotation/services/confrere.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Department } from '../../miscellaneous/model/Department';
import { Confrere } from '../../quotation/model/Confrere';

@Injectable({
  providedIn: 'root'
})
export class ConfrereService extends AppRestService<Confrere>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getConfreres() {
    return this.getList(new HttpParams(), "confreres");
  }

  getConfrereById(id: number) {
    return this.getById("confrere", id);
  }

  getConfrereFilteredByDepartmentAndName(department: Department | undefined, label: string) {
    let params = new HttpParams();
    if (department)
      params = params.set("departmentId", (department ? department.id : null) + "");
    return this.getList(params.set("label", label), "confreres/search");
  }

  addOrUpdateConfrere(confrere: Confrere) {
    this.clearListCache(new HttpParams(), "confreres");
    return this.addOrUpdate(new HttpParams(), "confrere", confrere, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/ConfrereService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.modules.osiris.miscellaneous.model.Department;
import com.jss.osiris.modules.osiris.quotation.model.Confrere;

public interface ConfrereService {
    public List<Confrere> getConfreres();

    public Confrere getConfrere(Integer id);

    public Confrere addOrUpdateConfrere(Confrere confrere) throws OsirisException;

    public List<Confrere> searchConfrereFilteredByDepartmentAndName(Department department, String label);
}
